src/main/java/net/coderbot/iris/pipeline/newshader/TriforcePatcher.java

```diff
@@ -151,13 +151,21 @@ public static String patchVanilla(String source, ShaderType type, AlphaTest alph
 
 		if (inputs.hasColor()) {
 			// TODO: Handle the fragment / geometry shader here
-			transformations.define("gl_Color", "(iris_Color * iris_ColorModulator)");
+			if (alpha == AlphaTests.VERTEX_ALPHA) {
+				transformations.define("gl_Color", "vec4((iris_Color * iris_ColorModulator).rgb, 1)");
```

Comment:
This makes me wonder... can `iris_ColorModulator.a` affect the alpha test threshold? Maybe we need to handle that.

Reply:
Sure, but it's the wild west out here in modding :P

Reply:
Fair, I'll put a multiply to handle that.

Reply:
Done.